From cba7ea789586afb2623cda1c6d7f829b6d1c453f Mon Sep 17 00:00:00 2001
From: Andrew Dodd <atd7@cornell.edu>
Date: Mon, 10 Dec 2012 08:26:54 -0500
Subject: [PATCH 03/43] DisplayDevice: Backwards compatibility with old EGL

From 4.1 to 4.2, the display subsystem was reworked to
use SurfaceTextureClient/BufferQueue instead of
FramebufferNativeWindow for the framebuffer itself.

Unfortunately, some legacy EGL libraries make assumptions
that any framebuffer device will be FramebufferNativeWindow.

These EGL libraries will fail when used in 4.2 as if the
framebuffer is not FramebufferNativeWindow, they will
try to dequeue more than one buffer at a time, which
will cause a hang of the graphics subsystem.

This allows use of FramebufferNativeWindow to keep
legacy EGL implementations happy.  Confirmed EGL
implementations that need this include but are
not limited to:
---
 libs/ui/FramebufferNativeWindow.cpp          | 2 --
 libs/ui/include/ui/FramebufferNativeWindow.h | 7 +++----
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/libs/ui/FramebufferNativeWindow.cpp b/libs/ui/FramebufferNativeWindow.cpp
index de5bf52f3..3a2b1b3d2 100644
--- a/libs/ui/FramebufferNativeWindow.cpp
+++ b/libs/ui/FramebufferNativeWindow.cpp
@@ -29,9 +29,7 @@
 
 #include <ui/ANativeObjectBase.h>
 #include <ui/Fence.h>
-#define INCLUDED_FROM_FRAMEBUFFER_NATIVE_WINDOW_CPP
 #include <ui/FramebufferNativeWindow.h>
-#undef INCLUDED_FROM_FRAMEBUFFER_NATIVE_WINDOW_CPP
 #include <ui/Rect.h>
 
 #include <EGL/egl.h>
diff --git a/libs/ui/include/ui/FramebufferNativeWindow.h b/libs/ui/include/ui/FramebufferNativeWindow.h
index 97137c3b6..3c2e70574 100644
--- a/libs/ui/include/ui/FramebufferNativeWindow.h
+++ b/libs/ui/include/ui/FramebufferNativeWindow.h
@@ -14,10 +14,6 @@
  * limitations under the License.
  */
 
-#ifndef INCLUDED_FROM_FRAMEBUFFER_NATIVE_WINDOW_CPP
-#warning "FramebufferNativeWindow is deprecated"
-#endif
-
 #ifndef ANDROID_FRAMEBUFFER_NATIVE_WINDOW_H
 #define ANDROID_FRAMEBUFFER_NATIVE_WINDOW_H
 
@@ -32,6 +28,9 @@
 #include <ui/ANativeObjectBase.h>
 #include <ui/Rect.h>
 
+#include <hardware/fb.h>
+#include <hardware/gralloc.h>
+
 #ifdef STE_SAMSUNG_HARDWARE
 #define NUM_FRAME_BUFFERS 3
 #else
-- 
2.11.0

